### DISCUSSION TITLE
Repository commit history - readability

### DISCUSSION THREAD

**Discussion by t-bltg** [FedeClaudi/Term.jl#111] [General]
Hey @FedeClaudi,

Regarding the history of the repo, I think [squashing commits](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/about-merge-methods-on-github#squashing-your-merge-commits) using the github interface when merging a PR is something you'd want to consider, especially since your package is starting to be used by other devs.

This keeps the repo history clean, see as an example what we do with `Unicodeplots.jl`: as single commit such as https://github.com/JuliaPlots/UnicodePlots.jl/commit/7ad058b1f99fa4f42d075d169c80507a4cca3ed8 squashing the 50+ commits of the PR https://github.com/JuliaPlots/UnicodePlots.jl/pull/217.

This avoids un-necessary merge commits and polluting the history with development details (reverts, ...), and still allow details to be accessed when inspecting the PR link added to the final commit message.

**Comment by FedeClaudi** [comment]
That's a very good point, will do moving forwards. Thank you.
